Story/RainMeadow.StoryHooks.cs

```diff
@@ -49,9 +52,108 @@ private void StoryHooks()
 
             On.Oracle.CreateMarble += Oracle_CreateMarble;
             On.Oracle.SetUpMarbles += Oracle_SetUpMarbles;
+            On.Oracle.SetUpSwarmers += Oracle_SetUpSwarmers;
+            On.OracleSwarmer.BitByPlayer += OracleSwarmer_BitByPlayer;
+            On.SLOracleSwarmer.BitByPlayer += SLOracleSwarmer_BitByPlayer;
+            On.CoralBrain.CoralNeuronSystem.PlaceSwarmers += OnCoralNeuronSystem_PlaceSwarmers;
+            IL.CoralBrain.CoralNeuronSystem.PlaceSwarmers += ILCoralNeuronSystem_PlaceSwarmers;
 
         }
 
+        //Only spawn if we own the room
+        private void OnCoralNeuronSystem_PlaceSwarmers(On.CoralBrain.CoralNeuronSystem.orig_PlaceSwarmers orig, CoralBrain.CoralNeuronSystem self)
+        {
+            if (OnlineManager.lobby == null)
+            {
+                orig(self);
+                return;
+            }
+            RoomSession.map.TryGetValue(self.room.abstractRoom, out var room);
+            if (room.isOwner)
+            {
+                orig(self);
+            }
+        }
+
+        private void ILCoralNeuronSystem_PlaceSwarmers(MonoMod.Cil.ILContext il)
+        {
+            try
+            {
+                // if (ModManager.MSC)
+                //becomes
+                // if (ModManager.MSC || OnlineManager.Lobby != null)
```

Comment:
Instead of "skipping a DLC check", I think we should either match `this.room.AddObject(ssoracleSwarmer);` OR maybe handle it from a regular hook to `ssoracleSwarmer.NewRoom(this.room)`

Reply:
addressed